ayup/portfolio/src/main/webapp/script.js

```diff
@@ -54,3 +54,62 @@ async function getRandomFactUsingAsyncAwait() {
   var a = document.getElementById('js-fact-container');
   a.parentNode.replaceChild(newFactContainer, a);
 }
+
+/** Fetches comments from the server and adds them to the DOM. */
+function loadComments() {
+  fetch('/list-comments').then(response => response.json()).then((comments) => {
+    const commentListElement = document.getElementById('comment-list');
+    console.log(comments)
+    comments.forEach((comment) => {
+      commentListElement.appendChild(createCommentElement(comment));
+    })
+  });
+}
+
+/** Creates an element that represents a title, a description, a username and its delete button. */
+function createCommentElement(comment) {
+  const commentElement = document.createElement('li');
+  commentElement.className = 'comment';
+
+  const titleElement = document.createElement('span');
+  titleElement.innerText = comment.title;
+  titleElement.className = "font-weight-bold";
+
+  const descriptionElement = document.createElement('span');
+  descriptionElement.innerText = comment.description;
+
+  const usernameElement = document.createElement('span');
+  usernameElement.innerText = "By: " + comment.username;
+
+  const deleteButtonElement = createDeleteButton()
+  deleteButtonElement.addEventListener('click', () => {
+    deleteComment(comment);
+
+    // Remove the comment from the DOM.
+    commentElement.remove();
```

Comment:
It would be better to re-fetch all comments from the server after deleting (what if deleting fails on the server?).

They mention that in the walkthrough https://github.com/googleinterns/step/blob/master/walkthroughs/week-3-server/step-6-additional-features-walkthrough.md#deleting-comments-1

Reply:
Done.